File: api/src/main/java/net/impactdev/gts/api/events/MaintenanceModeUpdateEvent.java
package net.impactdev.gts.api.events;

import net.impactdev.gts.api.maintenance.MaintenanceMode;
import net.impactdev.impactor.api.event.annotations.Param;

import java.util.UUID;

public interface MaintenanceModeUpdateEvent {

    @Param(0)
    UUID getServer();

    @Param(1)
    MaintenanceMode getMode();

    @Param(2)
    boolean getNewState();

}

File: api/src/main/kotlin/net/impactdev/gts/api/events/MaintenanceModeUpdateEvent.kt
package net.impactdev.gts.api.events

import net.impactdev.gts.api.maintenance.MaintenanceMode

interface MaintenanceModeUpdateEvent {
    @get:Param(0)
    val mode: MaintenanceMode?

    @get:Param(1)
    val newState: Boolean
}